### PR TITLE
Fix crash in SettingsViewModel when Vulkan isn't available

### LINKED ISSUE #4984: [Bug] `LoadAvailableGpus()` in `SettingsViewModel` can crash on weird configurations
### Description of the issue

In a Windows 11 for ARM Parallels VM on Apple Silicon, in which Vulkan is not an available API, attempting to open the Settings menu in Ryujinx.Ava will crash Ryujinx as the `Vk.GetApi()` call in `LoadAvailableGpus()` fails.

### Reproduction steps

- Open Ryujinx.Ava in Windows 11 Parallels VM
- Open settings menu
- Crash

### Log file

00:00:00.257 |I| Configuration LogValueChange: ResScale set to: 1
00:00:00.290 |I| Configuration LogValueChange: ResScaleCustom set to: 1
00:00:00.292 |I| Configuration LogValueChange: MaxAnisotropy set to: -1
00:00:00.296 |I| Configuration LogValueChange: AspectRatio set to: Fixed16x9
00:00:00.298 |I| Configuration LogValueChange: BackendThreading set to: Auto
00:00:00.302 |I| Configuration LogValueChange: GraphicsBackend set to: OpenGl
00:00:00.306 |I| Configuration LogValueChange: PreferredGpu set to: 
00:00:00.309 |I| Configuration LogValueChange: AntiAliasing set to: None
00:00:00.313 |I| Configuration LogValueChange: ScalingFilter set to: Bilinear
00:00:00.314 |I| Configuration LogValueChange: ScalingFilterLevel set to: 80
00:00:00.326 |I| Configuration LogValueChange: EnableDockedMode set to: True
00:00:00.372 |I| Configuration LogValueChange: EnableVsync set to: True
00:00:00.373 |I| Configuration LogValueChange: EnableShaderCache set to: True
00:00:00.374 |I| Configuration LogValueChange: EnableTextureRecompression set to: False
00:00:00.376 |I| Configuration LogValueChange: EnableMacroHLE set to: True
00:00:00.377 |I| Configuration LogValueChange: EnablePtc set to: True
00:00:00.378 |I| Configuration LogValueChange: EnableInternetAccess set to: False
00:00:00.378 |I| Configuration LogValueChange: EnableFsIntegrityChecks set to: True
00:00:00.379 |I| Configuration LogValueChange: FsGlobalAccessLogMode set to: 0
00:00:00.392 |I| Configuration LogValueChange: AudioBackend set to: SDL2
00:00:00.404 |I| Configuration LogValueChange: AudioVolume set to: 1
00:00:00.444 |I| Configuration LogValueChange: MemoryManagerMode set to: HostMappedUnsafe
00:00:00.448 |I| Configuration LogValueChange: ExpandRam set to: False
00:00:00.455 |I| Configuration LogValueChange: IgnoreMissingServices set to: False
00:00:00.457 |I| Configuration LogValueChange: UseHypervisor set to: True
00:00:00.501 |N| Application PrintSystemInfo: Ryujinx Version: 1.1.0+3e3b9f4
00:00:00.516 |N| Application Print: Operating System: Microsoft Windows 10.0.22621 (Arm64)
00:00:00.517 |N| Application Print: CPU: Virtual CPU @ 3.20GHz ; 6 logical
00:00:00.521 |N| Application Print: RAM: Total 12282 MiB ; Available 8915 MiB
00:00:00.529 |N| Application PrintSystemInfo: Logs Enabled: Info, Warning, Error, Guest, Stub
00:00:00.530 |N| Application PrintSystemInfo: Launch Mode: UserProfile
00:00:23.772 |E| Application : Unhandled exception caught: System.IO.FileNotFoundException: Could not load from any of the possible library names! Please make sure that the library is installed and in the right place!
   at Silk.NET.Vulkan.Vk.CreateDefaultContext(String[] n)
   at Silk.NET.Vulkan.Vk.GetApi()
   at Ryujinx.Ava.UI.ViewModels.SettingsViewModel.LoadAvailableGpus() in D:\a\Ryujinx\Ryujinx\src\Ryujinx.Ava\UI\ViewModels\SettingsViewModel.cs:line 314
   at Ryujinx.Ava.UI.ViewModels.SettingsViewModel..ctor() in D:\a\Ryujinx\Ryujinx\src\Ryujinx.Ava\UI\ViewModels\SettingsViewModel.cs:line 298
   at Ryujinx.Ava.UI.ViewModels.SettingsViewModel..ctor(VirtualFileSystem virtualFileSystem, ContentManager contentManager) in D:\a\Ryujinx\Ryujinx\src\Ryujinx.Ava\UI\ViewModels\SettingsViewModel.cs:line 275
   at Ryujinx.Ava.UI.Windows.SettingsWindow..ctor(VirtualFileSystem virtualFileSystem, ContentManager contentManager) in D:\a\Ryujinx\Ryujinx\src\Ryujinx.Ava\UI\Windows\SettingsWindow.axaml.cs:line 19
   at Ryujinx.Ava.UI.Views.Main.MainMenuBarView.OpenSettings(Object sender, RoutedEventArgs e) in D:\a\Ryujinx\Ryujinx\src\Ryujinx.Ava\UI\Views\Main\MainMenuBarView.axaml.cs:line 130
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object)
   at Avalonia.Threading.SendOrPostCallbackDispatcherOperation.InvokeCore() in /_/src/Avalonia.Base/Threading/DispatcherOperation.cs:line 392
   at Avalonia.Threading.DispatcherOperation.Execute() in /_/src/Avalonia.Base/Threading/DispatcherOperation.cs:line 268
   at Avalonia.Threading.Dispatcher.ExecuteJob(DispatcherOperation) in /_/src/Avalonia.Base/Threading/Dispatcher.Queue.cs:line 121
   at Avalonia.Threading.Dispatcher.ExecuteJobsCore() in /_/src/Avalonia.Base/Threading/Dispatcher.Queue.cs:line 176
   at Avalonia.Threading.Dispatcher.Signaled() in /_/src/Avalonia.Base/Threading/Dispatcher.Queue.cs:line 129
   at Avalonia.Win32.Win32Platform.WndProc(IntPtr, UInt32, IntPtr, IntPtr) in /_/src/Windows/Avalonia.Win32/Win32Platform.cs:line 219
   at Avalonia.Win32.Interop.UnmanagedMethods.DispatchMessage(MSG&)
   at Avalonia.Win32.Win32DispatcherImpl.RunLoop(CancellationToken) in /_/src/Windows/Avalonia.Win32/Win32DispatcherImpl.cs:line 111
   at Avalonia.Threading.DispatcherFrame.Run(IControlledDispatcherImpl) in /_/src/Avalonia.Base/Threading/DispatcherFrame.cs:line 117
   at Avalonia.Threading.Dispatcher.PushFrame(DispatcherFrame) in /_/src/Avalonia.Base/Threading/Dispatcher.MainLoop.cs:line 53
   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken) in /_/src/Avalonia.Base/Threading/Dispatcher.MainLoop.cs:line 80
   at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.Start(String[]) in /_/src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs:line 122
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime(AppBuilder, String[], ShutdownMode ) in /_/src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs:line 215
   at Ryujinx.Ava.Program.Main(String[] args) in D:\a\Ryujinx\Ryujinx\src\Ryujinx.Ava\Program.cs:line 51

### OS

Windows 11 for ARM

### Ryujinx version

1.1.800

### Game version

_No response_

### CPU

M1

### GPU

M1

### RAM

_No response_

### List of applied mods

_No response_

### Additional context?

Means I can't test how Ava PRs affect Windows effectively on my Mac.

### PR DESCRIPTION
**Pull request by jhorv**
Fixes #4984 
Doesn't make Ryujinx work on Win11 on ARM, because after this it crashes in ARMeilleure.

### CODE REVIEW

**Comment by IsaacMarovitz** [comment]
Thank you for fixing this very stupid bug lol

File: src/Ryujinx.Ava/UI/ViewModels/SettingsViewModel.cs [hunk 1]
```diff
@@ -311,7 +311,17 @@ private void LoadAvailableGpus()
...
+
+            var devices = Array.Empty<Ryujinx.Graphics.GAL.DeviceInfo>();
+            
+            try
+            {
+                devices = VulkanRenderer.GetPhysicalDevices(Vk.GetApi());
+            }
+            catch (Exception)
+            {
+                // Do nothing
```
Comment (AcK77) [inline src/Ryujinx.Ava/UI/ViewModels/SettingsViewModel.cs:323]: Maybe you should add a logger message here!

Comment (jhorv) [reply src/Ryujinx.Ava/UI/ViewModels/SettingsViewModel.cs]: Latest commit adds a parameterless overload to VulkanRenderer.GetPhysicalDevices() that catches/logs when Vk.GetApi() throws, and it adjusts the Ava and GTK front-ends to use this overload.

Comment (jhorv) [reply src/Ryujinx.Ava/UI/ViewModels/SettingsViewModel.cs]: Let me know if you think that's overkill. Simpler would be to try/catch around the 4 Vk.GetApi() call sites.

File: src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs [hunk 2]
```diff
@@ -600,6 +600,26 @@ public HardwareInfo GetHardwareInfo()
...
+        /// Gets the available Vulkan devices using the default Vulkan API
+        /// object returned by <see cref="Vk.GetApi()"/>
+        /// </summary>
+        /// <returns></returns>
+        public static DeviceInfo[] GetPhysicalDevices()
+        {
+            try
+            {
+                Vk api = Vk.GetApi();
+                return VulkanInitialization.GetSuitablePhysicalDevices(api);
```
Comment (AcK77) [inline src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs:613]:
```suggestion
                return VulkanInitialization.GetSuitablePhysicalDevices(Vk.GetApi());
```

File: src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs [hunk 3]
```diff
@@ -600,6 +600,26 @@ public HardwareInfo GetHardwareInfo()
...
+            }
+            catch (Exception ex)
+            {
+                // If we got an exception here, Vulkan is most likely not supported.
```
Comment (AcK77) [inline src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs:617]: Comment isn't really needed here.

File: src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs [hunk 4]
```diff
@@ -600,6 +600,26 @@ public HardwareInfo GetHardwareInfo()
...
+                Logger.Warning?.PrintMsg(LogClass.Gpu, $"Error querying Vulkan devices: {ex.Message}");
```
Comment (AcK77) [inline src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs:618]: Idk if we should use `Warning` or `Error` here

File: src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs [hunk 5]
```diff
@@ -600,6 +600,26 @@ public HardwareInfo GetHardwareInfo()
...
+                return Array.Empty<DeviceInfo>();
```
Comment (AcK77) [inline src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs:619]:
```suggestion

                return Array.Empty<DeviceInfo>();
```

**Review by AcK77 — CHANGES_REQUESTED** [review]

File: src/Ryujinx.Ava/UI/ViewModels/SettingsViewModel.cs (hunk 1 above)
Comment (AcK77) [reply src/Ryujinx.Ava/UI/ViewModels/SettingsViewModel.cs]: I prefer how it's done right now, then UIs doesnt have to manage anything ;)

File: src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs (hunk 3 above)
Comment (jhorv) [reply src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs]: Removed comment in commit [2b3cdcf](https://github.com/Ryujinx/Ryujinx/pull/4985/commits/2b3cdcf314e39d7c2192d92bb7d29aab5f93f943)

File: src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs (hunk 4 above)
Comment (jhorv) [reply src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs]: Made it use Error in commit [2b3cdcf](https://github.com/Ryujinx/Ryujinx/pull/4985/commits/2b3cdcf314e39d7c2192d92bb7d29aab5f93f943)

File: src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs (hunk 5 above)
Comment (jhorv) [reply src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs:618]: Fixed in commit [2b3cdcf](https://github.com/Ryujinx/Ryujinx/pull/4985/commits/2b3cdcf314e39d7c2192d92bb7d29aab5f93f943)

**Review by AcK77 — APPROVED** [review]

**Review by marysaka — APPROVED** [review]

File: src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs [hunk 6]
```diff
@@ -600,6 +600,25 @@ public HardwareInfo GetHardwareInfo()
             return new HardwareInfo(GpuVendor, GpuRenderer);
         }
 
+        /// <summary>
+        /// Gets the available Vulkan devices using the default Vulkan API
+        /// object returned by <see cref="Vk.GetApi()"/>
+        /// </summary>
+        /// <returns></returns>
+        public static DeviceInfo[] GetPhysicalDevices()
```
Comment (marysaka) [inline src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs:608]: Actually I would prefer Vk.GetApi to be passed by argument still.

**Review by marysaka — CHANGES_REQUESTED** [review]

File: src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs (hunk 6 above)
Comment (jhorv) [reply src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs:608]: > Actually I would prefer Vk.GetApi to be passed by argument still. (Required for some stuffs I have been working on)

There is still the overloaded version of `GetPhysicalDevices(Vk api)`. But are you saying you want me to remove the parameterless overload, and try/catch at the UI level?

Comment (AcK77) [reply src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs:608]: I don't think the try/catch should be done at the UI side.

**Review by marysaka — APPROVED** [review]

**Pull request merged**